plugin/src/main/java/org/opensearch/ml/model/MLModelAutoReLoader.java

```diff
@@ -0,0 +1,352 @@
+        threadPool.generic().submit(() -> {
+            try {
+                autoReLoadModelByNodeId(localNodeId);
+            } catch (ExecutionException | InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    /**
+     * auto reload all the models under the node id<br> the node must be a ml node<br>
+     *
+     * @param localNodeId node id
+     */
+    @VisibleForTesting
+    void autoReLoadModelByNodeId(String localNodeId) throws ExecutionException, InterruptedException {
+        StepListener<IndicesExistsResponse> indicesExistsResponseStepListener = new StepListener<>();
+        StepListener<SearchResponse> queryTaskStep = new StepListener<>();
+        StepListener<SearchResponse> getReTryTimesStep = new StepListener<>();
+        StepListener<IndexResponse> saveLatestReTryTimesStep = new StepListener<>();
+
+        indicesExistsResponseStepListener.whenComplete(indicesExistsResponse -> {
+            if (indicesExistsResponse.isExists()) {
+                queryTask(localNodeId, ActionListener.wrap(queryTaskStep::onResponse, queryTaskStep::onFailure));
+            }
+        }, indicesExistsResponseStepListener::onFailure);
+
+        isExistedIndex(ML_TASK_INDEX, indicesExistsResponseStepListener);
+
+        getReTryTimes(localNodeId, ActionListener.wrap(getReTryTimesStep::onResponse, getReTryTimesStep::onFailure));
+
+        Result result = new Result();
+        queryTaskStep.whenComplete(searchResponse -> {
+            SearchHit[] hits = searchResponse.getHits().getHits();
+            if (CollectionUtils.isEmpty(hits)) {
+                return;
+            }
+            result.setHits(hits);
+
+            getReTryTimesStep.whenComplete(getReTryTimesResponse -> {
+                // if getReTryTimesResponse is null,it means we get reTryTimes at the first time,and the index
+                // .plugins-ml-model-reload doesn't exist,so we should let reTryTimes be zero(init value)
+                // we don't do anything
+                // if getReTryTimesResponse is not null,it means we have saved the value of reTryTimes into the index
+                // .plugins-ml-model-reload,so we get the value of the field MODEL_LOAD_RETRY_TIMES_FIELD
+                if (getReTryTimesResponse != null) {
+                    Map<String, Object> sourceAsMap = getReTryTimesResponse.getHits().getHits()[0].getSourceAsMap();
+                    result.setReTryTimes((Integer) sourceAsMap.get(MODEL_LOAD_RETRY_TIMES_FIELD));
+                }
+
+                // According to the node id to get retry times, if more than the max retry times, don't need to retry
+                // that the number of unsuccessful reload has reached the maximum number of times, do not need to reload
+                if (result.getReTryTimes() > autoReLoadMaxReTryTimes) {
+                    log.info("have exceeded max retry times, always failure");
```

Comment:
How about: log.info("Node: {} has reached to the max retry limit, failed to load models", localNodeId)

Reply:
that's cool

Reply:
have committed the code